src/VisualStudio/Core/Def/Implementation/AddUsingsOnPaste/VisualStudioAddUsingsOnPaste.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable 
+
+using System;
+using System.Threading.Tasks;
+using System.Composition;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using System.Threading;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.VisualStudio.Utilities;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.AddMissingImports;
+using Microsoft.CodeAnalysis.AddMissingImports;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.AddUsingsOnPaste
+{
+    [ExportWorkspaceService(typeof(IAutomaticallyAddMissingUsingsService)), Shared]
+    internal class VisualStudioAddUsingsOnPaste : IAutomaticallyAddMissingUsingsService
+    {
+        private readonly IThreadingContext _threadingContext;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public VisualStudioAddUsingsOnPaste(
+            IThreadingContext threadingContext)
+        {
+            _threadingContext = threadingContext;
+        }
+
+        public void AddMissingUsings(Document document, TextSpan textSpan, IUIThreadOperationContext operationContext)
+        {
+            using var _ = operationContext.AddScope(true, ServicesVSResources.Adding_missing_import_statements);
```

Comment:
nit: label for `true`.. I don't know what it is... cancellable?

Reply:
*shakes fist at awesome features*

Reply:
Yeah, that's caught me a few times :)